app/controllers/automated_tests_controller.rb

```diff
@@ -14,48 +14,27 @@ def update
     @assignment = Assignment.find(params[:assignment_id])
     create_test_repo(@assignment)
 
-    # Perform transaction, if errors, none of new config saved
-    @assignment.transaction do
-      # Get new script from upload form
-      new_script = params[:new_script]
-      # Get new support file from upload form
-      new_support_file = params[:new_support_file]
-
-      @assignment = process_test_form(@assignment,
-                                      params,
-                                      assignment_params,
-                                      new_script,
-                                      new_support_file)
-      # Save assignment and associated test files
-      if @assignment.save
-        flash_message(:success, I18n.t('assignment.update_success'))
-        unless new_script.nil?
-          assignment_tests_path = File.join(
-              MarkusConfigurator.markus_ate_client_dir,
-              @assignment.repository_folder,
-              new_script.original_filename)
-          # Replace bad line endings from windows
-          contents = new_script.read.tr("\r", '')
-          File.open(
-              assignment_tests_path, 'w') { |f| f.write contents }
-        end
-
-        unless new_support_file.nil?
-          assignment_tests_path = File.join(
-              MarkusConfigurator.markus_ate_client_dir,
-              @assignment.repository_folder,
-              new_support_file.original_filename)
-          contents = new_support_file.read
-          File.open(
-              assignment_tests_path, 'wb') { |f| f.write contents }
+    begin
+      @assignment.transaction do
+        files = process_test_form(@assignment, params, assignment_params)
+        if @assignment.save
+          # write the uploaded files
+          files.each do |file|
+            File.open(file[:path], 'wb') { |f| f.write file[:upload].read }
+            if file.has_key?(:delete)
+              File.delete(file[:delete]) if File.exist?(file[:delete])
+            end
+          end
+          flash_message(:success, I18n.t('assignment.update_success'))
```

Comment:
You should be able to do this without the `I18n` now.

Reply:
Do you mean there is a default message for a :success?

Reply:
No I mean you should be able to use `t(...)` rather than `I18n.t(...)` in all controllers now.